cartographer_ros/cartographer_ros/node_main.cc

```diff
@@ -80,17 +80,31 @@ void Run() {
 }  // namespace cartographer_ros
 
 int main(int argc, char** argv) {
-  // It is important to initialize rclcpp first or copy argv,
-  // because the google commands eat the arguments
-  ::rclcpp::init(argc, argv);
-  google::InitGoogleLogging(argv[0]);
-  google::ParseCommandLineFlags(&argc, &argv, true);
+  // Strip ROS specific arguments
```

Comment:
As we discussed in person, GFLAGS also has the `--undefok` [flag](https://gflags.github.io/gflags/#special).  However, we would have to enumerate *all* ROS flags there, which is bad.  Shane found an alternate workaround that we are going to try: https://github.com/gflags/gflags/issues/148#issuecomment-318826625